forge-game/src/main/java/forge/game/ability/effects/CountersPutAllEffect.java

```diff
@@ -71,6 +71,9 @@ public void resolve(SpellAbility sa) {
         GameEntityCounterTable table = new GameEntityCounterTable();
         for (final Card tgtCard : cards) {
             tgtCard.addCounter(type, counterAmount, placer, table);
+            if (sa.hasParam("RememberCards")) {
+                host.addRemembered(tgtCard);
```

Comment:
Not good enough, some counters might be prevented/replaced to 0

Maybe it'll fit inside `replaceCounterEffect`

Reply:
The problem is even if all get replaced the table will still contain the card entries :/
(just tested here, the Ant still got goaded even though those Voris stopped the counters)
![image](https://user-images.githubusercontent.com/8506892/181116736-12edf145-9414-44bf-af28-dc26a25c76ca.png)

That's why I meant you need to check the values deeper inside there
(it does look like some other cards may use RememberCards$ differently, that needs to be sorted out or renamed?)

Reply:
Yeah, I modeled this new attempt after what I saw in CountersPutEffect

We could both of them into replaceCounterEffect as you suggest

Reply:
Yea, maybe something like "RememberPut" for the Ant and this

The way it's done currently we still need the  `RememberCards` for stuff like _Thought Gorger_.

That is because of CR 118.12:
> The "If [a player] [does, doesn't, or can't]" clause checks whether the player chose to pay an optional cost or started to pay a mandatory cost, regardless of what events actually occurred.

That means if _Solemnity_ is in play you can't even start to pay, so no discard will happen

On the other hand if the counters just get replaced, it should still be satisfied

Reply:
New code passes your Vorinclex/Helm of the Host test – see what you think

Reply:
Yea should be good :)